libtock/alarm_timer.c

```diff
@@ -100,8 +101,12 @@ void alarm_at(uint32_t reference, uint32_t dt, subscribe_upcall cb, void* ud, al
 
   if (root_peek() == alarm) {
     alarm_internal_subscribe((subscribe_upcall*)callback, NULL);
-    alarm_internal_set(alarm->reference, alarm->dt);
+
+    if (alarm_internal_set(alarm->reference, alarm->dt) != RETURNCODE_SUCCESS) {
+      return RETURNCODE_FAIL;
```

Comment:
Can't you just directly `return alarm_internal_set(...`?

Reply:
Can do, just wondering in the case the syscall fails, this can return a status-code from the kernel (still < 0), from `tock_status_to_returncode()` to `tock_command_return_u32_to_returncode()` back to  `alarm_internal_set()` and out.  Not sure if there's a preference to how this should be handled?